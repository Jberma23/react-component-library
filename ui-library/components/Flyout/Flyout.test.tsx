import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Flyout } from "@ui-library/index";

describe(Flyout, () => {
  it("matches the snapshot with defaults", () => {
    const { asFragment } = render(
      <Flyout>
        <div>Hello world</div>
      </Flyout>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches the snapshot when closed", () => {
    const { asFragment } = render(
      <Flyout fillScreen isOpen={false} openFromDirection={"left"}>
        Hello World
      </Flyout>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches the snapshot when open", () => {
    const { asFragment } = render(
      <Flyout fillScreen isOpen openFromDirection={"left"}>
        Hello World
      </Flyout>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("is visible to screen readers when open", () => {
    render(
      <Flyout fillScreen isOpen openFromDirection={"left"}>
        Hello World
      </Flyout>
    );
    expect(screen.getByText("Hello World")).toBeInTheDocument();
  });

  it("passes a11y checks when open", async () => {
    await expect(
      <Flyout fillScreen isOpen openFromDirection={"left"}>
        Hello World
      </Flyout>
    ).toBeA11yWhenRendered();
  });

  it("will call the onClose method when closeOnOverlayClick is true and user clicks the overlay", async () => {
    const onClose = jest.fn();
    render(
      <Flyout
        fillScreen
        onClose={onClose}
        openFromDirection={"left"}
        closeOnOverlayClick
        showOverlay
      >
        Hello World
      </Flyout>
    );

    await userEvent.click(
      document.querySelector<HTMLElement>(".flyout-overlay")!
    );

    expect(onClose).toHaveBeenCalled();
  });
});
