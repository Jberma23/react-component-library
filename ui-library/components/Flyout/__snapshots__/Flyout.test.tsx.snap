// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout matches the snapshot when closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed top-[64px] left-0 right-0 bottom-0 z-50 pointer-events-none"
  >
    <div
      class="fixed h-[calc(100%-64px)] bg-white z-20 overflow-auto shadow-1 duration-default transition-all left-0 w-screen h-screen -translate-x-[150%]"
    />
    <div
      aria-hidden="true"
      class="flyout-overlay fixed w-full h-full inset-0 transition-opacity duration-default bg-black/50 opacity-0 pointer-events-none z-10"
    />
  </div>
</DocumentFragment>
`;

exports[`Flyout matches the snapshot when open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="fixed top-[64px] left-0 right-0 bottom-0 z-50 pointer-events-auto"
  >
    <div
      class="fixed h-[calc(100%-64px)] bg-white z-20 overflow-auto shadow-1 duration-default transition-all left-0 w-screen h-screen translate-x-[0%]"
    >
      Hello World
    </div>
    <div
      aria-hidden="false"
      class="flyout-overlay fixed w-full h-full inset-0 transition-opacity duration-default bg-black/50 pointer-events-auto z-10"
    />
  </div>
</DocumentFragment>
`;

exports[`Flyout matches the snapshot with defaults 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed top-[64px] left-0 right-0 bottom-0 z-50 pointer-events-none"
  >
    <div
      class="fixed h-[calc(100%-64px)] bg-white z-20 overflow-auto shadow-1 duration-default transition-all left-0 min-w-full md:min-w-[350px] min-h-[200px] -translate-x-[150%]"
    />
    <div
      aria-hidden="true"
      class="flyout-overlay fixed w-full h-full inset-0 transition-opacity duration-default bg-black/50 opacity-0 pointer-events-none z-10"
    />
  </div>
</DocumentFragment>
`;
